### PR TITLE
chore: change select Zoe methods to take a feePurse

### LINKED ISSUE #3309: Add feePurse
A feePurse (formerly "chargeAccount") is a representation of RUN escrowed in Zoe. A feePurse can be created by:

```js
const feePurse = E(zoe).makeFeePurse();
```

The purse is created within Zoe. The purse is synchronously accessible within Zoe and asynchronously accessible outside. Zoe can be sure that when it calls `feePurse.getCurrentAmount()` and then withdraws an amount less than the current amount, the withdrawal will succeed without interference. The user cannot be sure of this, as the user's wallet can only call asynchronously, and the amount may have changed between the query for the balance and the actual withdrawal.

The feePurse has the methods of an ERTP purse for RUN, but is specially recognized by Zoe. Zoe does not accept purses generally, because it has no way of verifying a purse made by a user.

### LINKED ISSUE #3322: Add feePurse arg to Zoe methods. Also partially apply the feePurse to produce original Zoe API
As part of #3294, we should add the feePurse to be used for paying for execution, to Zoe methods. We can separate this task from the actual charging of the feePurse. In effect, feePurses are required, but all calls are still free.

If we assume a lower-level stamp model (see #3294) that reduces spam generally, the only methods that require a feePurse are:
```js
    install,
    startInstance,
    offer,
    getPublicFacet
```

And the following are queries that only require a stamp and no charge:

```js
    getInvitationIssuer,
    getBrands,
    getIssuers,
    getTerms,
    getInstance,
    getInstallation,
    getInvitationDetails,
```

`makeFeePurse` is not a query, but it's also no charge because to charge a fee for making a feePurse wouldn't be very sensical. 

As part of this task, we should provide a version of Zoe that has the original API, with the feePurse partially applied. This will allow us to retain the old tests mostly unchanged, which lightens the load significantly.

### PR DESCRIPTION
**Pull request by katelynsills**
Changes select Zoe methods to take a feePurse. Provides a function to apply a fee purse to Zoe to get the original "ZoeService" API. 

Closes #3322

Stacked on https://github.com/Agoric/agoric-sdk/pull/3652

### CODE REVIEW

File: packages/zoe/src/applyFeePurse.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ * @param {ERef<ZoeServiceFeePurseRequired>} zoe
+ * @param {ERef<FeePurse>} feePurse
+ * @returns {ZoeService}
+ */
+const applyFeePurse = (zoe, feePurse) => {
+  return Far('zoeService', {
+    makeFeePurse: (...args) => E(zoe).makeFeePurse(...args),
+
+    // A feePurse is required
+    install: (bundle, _overriddenFeePurse) => E(zoe).install(bundle, feePurse),
```
Comment (erights) [inline packages/zoe/src/applyFeePurse.js]: What I had in mind was more like the following pattern

```suggestion
const applyFeePurse = (zoe, defaultFeePurse) => {
  return Far('zoeService', {
    makeFeePurse: (...args) => E(zoe).makeFeePurse(...args),

    // A feePurse is required
    install: (bundle, feePurse = defaultFeePurse) => E(zoe).install(bundle, feePurse),
```

where the bound purse is the default, and the caller can override the default on any individual call if they provide an explicit argument.

File: packages/zoe/src/contracts/loan/lend.js [hunk 2]
```diff
@@ -38,7 +38,7 @@ export const makeLendInvitation = (zcf, config) => {
       exit: { onDemand: null }, // The lender must be able to exit with their loan at any time before the money is borrowed
     });
 
-    const configWithLenderSeat = { ...config, lenderSeat };
+    const configWithLenderSeat = { ...config, ...privateArgs, lenderSeat };
```
[the remark this answers is not in the document]
Comment (katelynsills) [reply packages/zoe/src/contracts/loan/lend.js]: Ah, I think this comment is outdated

File: packages/zoe/src/applyFeePurse.js (hunk 1 above)
Comment (katelynsills) [reply packages/zoe/src/applyFeePurse.js]: Ah yep, sorry, I was getting too tired while doing this. Your suggestion is the only way this would work.

Comment (katelynsills) [reply packages/zoe/src/applyFeePurse.js]: Ok, that should be fixed now with tests as well!

**Review by erights — COMMENTED** [review]
I tried an experiment in https://github.com/Agoric/agoric-sdk/pull/3675

Seems successful enough to consider for revising this PR.

**Review by erights — APPROVED** [review]
LGTM!

**Pull request merged**